src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java

```diff
@@ -178,15 +187,24 @@ public Map<String, Object> getUserProgress(SunbirdApiRequest requestBody, String
 				userEnrolmentList = userEnrolmentList.subList(0, 100);
 			}
 
-			// get all user details
-			List<String> enrolledUserId = userEnrolmentList.stream()
-					.map(obj -> (String) obj.get(Constants.USER_ID_CONSTANT)).collect(Collectors.toList());
 			List<String> userFields = new ArrayList<>(Arrays.asList(Constants.USER_ID_CONSTANT, Constants.FIRSTNAME, Constants.PROFILE_DETAILS_PRIMARY_EMAIL, Constants.CHANNEL,
 					Constants.PROFILE_DETAILS_DESIGNATION, Constants.PROFILE_DETAILS_DESIGNATION_OTHER));
-			Map<String, Object> userMap = userUtilService.getUsersDataFromUserIds(enrolledUserId, userFields,
+			Map<String, Object> userMap = userUtilService.getUsersDataFromUserIds(enrollmentIdList, userFields,
 					authUserToken);
 
 			Map<String, Integer> courseLeafCount = new HashMap<>();
+			if (!courseLeafCount.containsKey(requestData.getCourseId())) {
```

Comment:
This should be outside of the getBatchList for loop.

Reply:
Changes made as suggested @karthik-tarento